test/integration/scheduler/queue_test.go

```diff
@@ -748,28 +865,115 @@ func TestCoreResourceEnqueue(t *testing.T) {
+				// It causes target-pod to be requeued.
+				pod := st.MakePod().Name("trigger-pod").Container("image").Obj()
+				if _, err := testCtx.ClientSet.CoreV1().Pods(testCtx.NS.Name).Create(testCtx.Ctx, pod, metav1.CreateOptions{}); err != nil {
+					return fmt.Errorf("failed to create pod: %w", err)
+				}
+				return nil
+			},
+			customPlugins: []string{"UnscheduledPodAdd"},
+			// trigger-pod is not actually requeued, but stayed in the active queue at first
+			wantRequeuedPodsByQHint: map[bool]sets.Set[string]{
+				false: sets.New("target-pod", "trigger-pod"),
+				true:  sets.New("target-pod", "trigger-pod"),
+			},
+		},
+		{
+			name:         "Pod rejected by the UnscheduledPodUpdate plugin is requeued when a UnscheduledPod is updated",
+			initialNodes: []*v1.Node{st.MakeNode().Name("fake-node").Obj()},
+			pods: []*v1.Pod{
+				// both pod will be rejected by the UnscheduledPodAdd plugin.
+				st.MakePod().Name("target-pod").Container("image").Obj(),
+				st.MakePod().Name("trigger-pod").Container("image").Obj(),
+			},
+			triggerFn: func(testCtx *testutils.TestContext) error {
+				// Trigger a UnscheduledPodAdd event.
+				// It causes target-pod to be requeued.
+				pod := st.MakePod().Name("trigger-pod").Container("image").Labels(map[string]string{
+					"foo": "bar",
+				}).Obj()
+				if _, err := testCtx.ClientSet.CoreV1().Pods(testCtx.NS.Name).Update(testCtx.Ctx, pod, metav1.UpdateOptions{}); err != nil {
+					return fmt.Errorf("failed to update pod: %w", err)
+				}
+				return nil
+			},
+			customPlugins: []string{"UnscheduledPodUpdate"},
+			// trigger-pod is not actually requeued, but stayed in the active queue at first
+			wantRequeuedPodsByQHint: map[bool]sets.Set[string]{
+				// when Qhint is disabled, any updates on a pod will make the pod requeued
+				false: sets.New("target-pod", "trigger-pod"),
+				true:  sets.New("target-pod"),
+			},
+		},
+		{
+			name:         "Pod rejected by the UnscheduledPodDelete plugin is requeued when a UnscheduledPod is deleted",
+			initialNodes: []*v1.Node{st.MakeNode().Name("fake-node").Obj()},
+			pods: []*v1.Pod{
+				// both pod will be rejected by the UnscheduledPodAdd plugin.
+				st.MakePod().Name("target-pod").Container("image").Obj(),
+				st.MakePod().Name("trigger-pod").Container("image").Obj(),
+			},
+			triggerFn: func(testCtx *testutils.TestContext) error {
+				// Trigger a UnscheduledPodAdd event.
+				// It causes target-pod to be requeued.
+				if err := testCtx.ClientSet.CoreV1().Pods(testCtx.NS.Name).Delete(testCtx.Ctx, "trigger-pod", metav1.DeleteOptions{}); err != nil {
+					return fmt.Errorf("failed to delete pod: %w", err)
+				}
+				return nil
+			},
+			customPlugins: []string{"UnscheduledPodDelete"},
+			wantRequeuedPodsByQHint: map[bool]sets.Set[string]{
+				false: sets.New("target-pod"),
+				true:  sets.New("target-pod"),
+			},
 		},
 	}
 
 	for _, tt := range tests {
-		if len(tt.enableSchedulingQueueHint) == 0 {
```

Comment:
To prevent a mistake, can we make the test fail when `len(tt.wantRequeuedPodsByQHint) == 0`?